tenpy/simulations/time_evolution.py

```diff
@@ -74,3 +80,347 @@ def final_measurements(self):
+            ops = ops * len(idx)
+        # generate list of tuples of form [(op1, i_1), (op2, i_2), ...]
+        op_list = list(zip(ops, idx))
+        return op_list
+
+    def apply_operator_t0_to_psi(self):
+        # TODO: think about segment boundary conditions
+        # TODO: make JW string consistent, watch for changes in apply_local_op to have autoJW
+        self.operator_t0 = self._get_operator_t0()
+        operator_t0 = self.operator_t0
+        if len(operator_t0) == 1:
+            op, i = operator_t0[0]
+            if self.model.lat.site(i).op_needs_JW(op):
+                for j in range(i):
+                    self.psi.apply_local_op(j, 'JW')
+            self.psi.apply_local_op(i, op)  # TODO: check if renormalize=True makes sense here
+        else:
+            ops, i_min, _ = self.psi._term_to_ops_list(operator_t0, True)  # applies JW string automatically
+            for i, op in enumerate(ops):
+                self.psi.apply_local_op(i_min + i, op)
+
+    def prepare_results_for_save(self):
+        """Wrapper around :meth:`prepare_results_for_save` of :class:`Simulation`.
+        Makes it possible to include post-processing run during the run of the
+        actual simulation.
+        """
+        if self.post_processor is not None:
+            self.logger.info(f"calling post-processing with {self.post_processor}")
+            processing_params = self.options.get('post_processing_params', None)
+            # try, except clause to not lose simulation results if post_processing fails
+            try:
+                post_processor_cls = self.post_processor.from_simulation(self, processing_params=processing_params)
+                # TODO: make sure this is written into self.results
+                post_processor_cls.run()
+            except Exception as e:
+                self.logger.info("Could not post-process the results because of the following exception:")
+                self.logger.warning(e)
+                self.logger.info("continuing saving results without post-processing")
+        return super().prepare_results_for_save()
+
+    def get_mps_environment(self):
+        return MPSEnvironment(self.psi_ground_state, self.psi)
+
+    def m_spectral_function(self, results, psi, model, simulation, **kwargs):
+        """Calculate the overlap :math:`<psi_0| e^{iHt} op2^j e^{-iHt} op1_idx |psi_0>` between
+        op1 at MPS position idx and op2 at the MPS position j"""
+        self.logger.info("calling m_spectral_function")
+        operator_t = self.operator_t
+        env = self.get_mps_environment()  # custom method for subclass Experimental
+        # TODO: get better naming convention, store this in dict ?
+        if isinstance(operator_t, list):
+            for i, op in enumerate(operator_t):
+                if isinstance(op, str):
+                    results[f'spectral_function_t_{op}'] = self._m_spectral_function_op(env, op)
```

Comment:
Yes, we need better naming conventions here. 
In general, I think we should allow more generic operator terms to be measured (as we discussed with Frank as well), we want generic `<psi | A(t) B|psi>`, with A/B sometimes being multi-site operators, see also the comment on `apply_operator_t0_to_psi` above.
I think a convenient parameter conventione might be: either `operator_t` is a single dict specifying the operator / term A (with MPS /lattice index and op name), or you have a dict `{opname_1: {...}, opname_2: {...}}`, which each of the sub dictionaries as you would have for a single operator.
Of course, the same convention for specifying a single term/local operator should be used for the both the t0 and t operators.